### PR TITLE
Match type email

### PR DESCRIPTION
**Pull request by RavirajBaraiya**
issue #141

### CODE REVIEW

File: core/src/main/java/zingg/similarity/function/EmailMatchTypeFunction.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
+package zingg.similarity.function;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+
+
+public class EmailMatchTypeFunction extends StringSimilarityFunction {
```
Comment (sonalgoyal) [inline core/src/main/java/zingg/similarity/function/EmailMatchTypeFunction.java:8]: extend StringSimilarityDistanceFunction and use that

File: core/src/test/java/zingg/similarity/function/TestEmailMatchTypeFunction.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+import java.util.Arrays;
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class TestEmailMatchTypeFunction {
+	
+	
+	@Test
```
Comment (sonalgoyal) [inline core/src/test/java/zingg/similarity/function/TestEmailMatchTypeFunction.java:13]: please update test scores asserts accordingly. you can invoke the StringSimilarityDistanceFunction with the first part of email and check that you are getting what it returns

**Review by sonalgoyal — CHANGES_REQUESTED** [review]

File: core/src/test/java/zingg/similarity/function/TestEmailMatchTypeFunction.java (hunk 2 above)
Comment (RavirajBaraiya) [reply core/src/test/java/zingg/similarity/function/TestEmailMatchTypeFunction.java:13]: I tried to do that. but, It is giving me this error: Cannot instantiate the type StringSimilarityDistanceFunction
should I leave that part by checking scores with EmailMatchTypeFunction ?

Comment (sonalgoyal) [reply core/src/test/java/zingg/similarity/function/TestEmailMatchTypeFunction.java:13]: ok, sorry I realized later it is an abstract class. Can you please extend AffineGapSimilarityFunction and use that in the tests as well?

Comment (RavirajBaraiya) [reply core/src/test/java/zingg/similarity/function/TestEmailMatchTypeFunction.java:13]: Ok, let me try that.

File: core/src/test/java/zingg/similarity/function/TestEmailMatchTypeFunction.java [hunk 3]
```diff
@@ -12,61 +12,101 @@ public class TestEmailMatchTypeFunction {
...
 	@Test
 	public void testFirstEntryNull() {
-		EmailMatchTypeFunction emailMatchFn = new EmailMatchTypeFunction();
-		assertEquals(1d, emailMatchFn.call(null, "xyz321@pqr.co"));
+		AffineGapSimilarityFunction emailMatchFn = new AffineGapSimilarityFunction();
+		String first = null;
+		String second = "xyz321@pqr.co";
+		//first = first.split("@",0)[0];
+		second = second.split("@",0)[0];
+		assertEquals(1d, emailMatchFn.call(first, second));
```
Comment (sonalgoyal) [inline core/src/test/java/zingg/similarity/function/TestEmailMatchTypeFunction.java:20]: you need to verify here that the output from emailmatchfn is the same as the output from first part of emails sent to affinegap fn

File: core/src/test/java/zingg/similarity/function/TestEmailMatchTypeFunction.java [hunk 4]
```diff
@@ -12,61 +12,101 @@ public class TestEmailMatchTypeFunction {
...
 
 	@Test
 	public void testFirstEntryEmpty() {
 		EmailMatchTypeFunction emailMatchFn = new EmailMatchTypeFunction();
-		assertEquals(1d, emailMatchFn.call("", "xyz321@pqr.co"));
+		String first = "";
+		String second = "xyz321@pqr.co";
+		first = first.split("@",0)[0];
+		second = second.split("@",0)[0];
+		assertEquals(1d, emailMatchFn.call(first, second));
```
Comment (sonalgoyal) [inline core/src/test/java/zingg/similarity/function/TestEmailMatchTypeFunction.java:30]: the function does the split here, you should not be splitting unless you want to send the split values to AffineGapSimilarityFunction 

so validate that original first and second when passed to emailfn returns the same as first part of split string sent to affinegap similarity, ok?

**Review by sonalgoyal — CHANGES_REQUESTED** [review]
test cases need a relook.

**Pull request merged**